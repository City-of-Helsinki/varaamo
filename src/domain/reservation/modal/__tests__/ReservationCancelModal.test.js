import React from 'react';
import toJSON from 'enzyme-to-json';

import { shallowWithIntl, globalDateMock } from '../../../../../app/utils/testUtils';
import { UnconnectedReservationCancelModal } from '../ReservationCancelModal';
import reservation from '../../../../common/data/fixtures/reservation';

describe('domain/reservation/modal/ReservationCancelModal', () => {
  globalDateMock();
  test('renders correctly', () => {
    const mockReservation = reservation.build({ begin: '2019-08-14T14:00:00+03:00', end: '2019-08-14T15:00:00+03:00' });
    const mockBoolean = true;
    const extraProps = {
      t: jest.fn(),
      userId: 'foo',
      users: { bar: 'bar' },
    };
    const props = {
      onEditReservation: jest.fn(),
      parentToggle: jest.fn(),
      reservation: mockReservation,
      toggleShow: mockBoolean,
    };
    const wrapper = shallowWithIntl(
      <UnconnectedReservationCancelModal {...props} {...extraProps} />,
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
