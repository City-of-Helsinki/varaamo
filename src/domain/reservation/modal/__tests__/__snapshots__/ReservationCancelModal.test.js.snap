// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`domain/reservation/modal/ReservationCancelModal renders correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropClassName="modal-backdrop"
  backdropTransition={[Function]}
  containerClassName="modal-open"
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": [],
      "data": [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": [],
      "remove": [Function],
    }
  }
  onEntering={[Function]}
  onExited={[Function]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
  transition={[Function]}
>
  <ModalDialog
    bsClass="modal"
    className=""
    onClick={[Function]}
    style={{}}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      />
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <div>
        <p>
          <strong />
        </p>
        <CompactReservationList
          reservations={
            [
              {
                "begin": "2019-08-14T14:00:00+03:00",
                "comments": "",
                "end": "2019-08-14T15:00:00+03:00",
                "event_description": "",
                "event_subject": "",
                "has_catering_order": false,
                "id": 1,
                "its_own": false,
                "resource": undefined,
                "staff_event": false,
                "state": "requested",
                "url": "https://respa.koe.hel.ninja/v1/reservation/192388/",
                "user": {},
                "user_permissions": null,
              },
            ]
          }
          resources={
            {
              "undefined": {
                "name": "",
              },
            }
          }
        />
      </div>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[Function]}
      />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="danger"
        disabled={false}
        onClick={[Function]}
      />
    </ModalFooter>
  </ModalDialog>
</Modal>
`;
